drivers/storage/portworx/deployment.go

```diff
@@ -318,11 +320,46 @@ func (p *portworx) GetStoragePodSpec(
 	}
 
 	if cluster.Spec.CloudStorage != nil && len(cluster.Spec.CloudStorage.CapacitySpecs) > 0 {
-		// Cloud Environment
-		cloudConfig, err := p.generateCloudStorageSpecs(cluster)
+		var cloudConfig *cloudstorage.Config
+		storageNodes, err := p.getStorageNodes()
 		if err != nil {
 			return v1.PodSpec{}, err
 		}
+		if len(storageNodes.Items) == 0 {
+			cloudConfig, err = p.generateCloudStorageSpecs(cluster)
+			if err != nil {
+				logrus.Error(err)
+				return v1.PodSpec{}, err
+			}
+
+			nodes, err := p.getNodesList()
+			if err != nil {
+				return v1.PodSpec{}, err
+			}
+
+			for _, node := range nodes.Items {
+				ownerRef := metav1.NewControllerRef(cluster, controllerKind)
+				storageNode := &corev1alpha1.StorageNode{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:            node.Name,
+						Namespace:       cluster.Namespace,
+						OwnerReferences: []metav1.OwnerReference{*ownerRef},
+						Labels:          p.GetSelectorLabels(),
+					},
+					Status: corev1alpha1.NodeStatus{},
+				}
+
+				configureStorageNodeSpec(storageNode, cloudConfig)
+				err := k8s2.CreateOrUpdateStorageNode(p.k8sClient, storageNode, ownerRef)
+				if err != nil {
+					msg := fmt.Sprintf("Failed to create node for nodeID %v: %v", node.Name, err)
+					p.warningEvent(cluster, failedSyncReason, msg)
+				}
+			}
+		} else {
+			// Cloud Environment
```

Comment:
nit: This comments needs to be at the start of the outer if condition where it was before.